kuka_ros2_control_support/test/ros2_control_support.py

```diff
@@ -179,16 +187,45 @@ class RobotDriverTest(unittest.TestCase):
     def setUpClass(cls):
         # Initialize the ROS context
         rclpy.init()
+
         cls.node_name = "ros2_control_support_test"
         cls.node = Node(cls.node_name)
-        time.sleep(1)
+        cls.executor = rclpy.executors.SingleThreadedExecutor()
+        cls.trajectory_publisher = cls.node.create_publisher(
+            JointTrajectory, "/position_trajectory_controller/joint_trajectory", 10
+        )
+        cls.node.get_logger().info("Setup complete")
+
+    @classmethod
+    def wait_for_trajectory_state(cls, time_to_wait=3.0):
+        ret, msg = wait_for_message(
+            JointTrajectoryControllerState,
+            cls.node,
+            "/position_trajectory_controller/state",
+            time_to_wait=time_to_wait,
+        )
+        cls.assertTrue(cls, ret)
```

Comment:
```suggestion
        cls.assertTrue(ret, "failed to receive a trajectory state message")
```
The problem is, that with no message, the assertion fails but the code continues and then throw errors due to msg being None. If one protects it from failing, later code expects the joint_states to be set and fails then later.

Reply:
thanks